### PR TITLE
Port GifControlDirectory additions from Java

### PR DESCRIPTION
**Pull request by kwhopper**
Todo: backport GifControlDescriptor "{value}" addition to Java
Todo: backport to Java GifReader --  directory.Set(GifControlDirectory.TagUserInputFlag, (packedFields & 2) == 2);

### CODE REVIEW

File: MetadataExtractor/Formats/Gif/GifReader.cs [hunk 1]
```diff
@@ -329,12 +329,12 @@ private static GifControlDirectory ReadControlBlock(SequentialReader reader, byt
...
-
+            byte packedFields = reader.GetByte();
+            directory.Set(GifControlDirectory.TagDisposalMethod, (packedFields >> 2) & 7);
+            directory.Set(GifControlDirectory.TagUserInputFlag, (packedFields & 2) == 2);
+            directory.Set(GifControlDirectory.TagTransparentColorFlag, (packedFields & 1) == 1);
             directory.Set(GifControlDirectory.TagDelay, reader.GetUInt16());
-
-            if (blockSizeBytes > 3)
-                reader.Skip(blockSizeBytes - 3);
+            directory.Set(GifControlDirectory.TagTransparentColorIndex, reader.GetByte());
```
Comment (drewnoakes) [inline MetadataExtractor/Formats/Gif/GifReader.cs]: This used to sometimes skip three bytes, but now always consumes only one. Is this correct?

Comment (kwhopper) [reply MetadataExtractor/Formats/Gif/GifReader.cs]: I don't know for sure without further investigation. This was ported directly from the Java version approved through drewnoakes/metadata-extractor#241.

Maybe @Nadahar has some info on it?

Comment (Nadahar) [reply MetadataExtractor/Formats/Gif/GifReader.cs]: I'm sorry, I checked in my memory, but I'm unable to revive any information about this. From what I can see from the PR, additional parsing was added. It doesn't seem unreasonable that less will be skipped when more is parsed.. does this behavior cause a problem?

Comment (drewnoakes) [reply MetadataExtractor/Formats/Gif/GifReader.cs]: > additional parsing was added

It's just unpacking a single byte that was previously skipped.

@kwhopper if you're happy that your testing produced better results with this change then I'm happy.

Comment (kwhopper) [reply MetadataExtractor/Formats/Gif/GifReader.cs]: It appears to produce the same results as the Java implementation against the image repository. To me, that's the best we can reasonably export when porting.

Comment (drewnoakes) [reply MetadataExtractor/Formats/Gif/GifReader.cs]: Thank you for confirming. Merged!

**Comment by drewnoakes** [comment]
> Todo: backport GifControlDescriptor "{value}" addition to Java
> Todo: backport to Java GifReader -- directory.Set(GifControlDirectory.TagUserInputFlag, (packedFields & 2) == 2);

Could you open issue for these two please so they're not lost?

**Pull request merged**